alimasa_eliza/countries_script.js

```diff
@@ -0,0 +1,62 @@
+function countrySearch() {
+    const countryName = document.getElementById('country_input').value;
+    if (!countryName) {
+        alert('Please enter a country name.');
+        return;
+    }
+    
+    fetch('https://restcountries.com/v3.1/name/' + countryName)
+    .then(function(response) {
+      if (!response.ok) {
+        throw new Error('Country not found');
+      }
+      return response.text();
+      })
+    .then(function(countryData) {
+    let country = JSON.parse(countryData)[0];
+    let details = `
+     <h2>${country.name.common}</h2>
+     Population: ${country.population.toLocaleString()}
+     <p>Area: ${country.area ? country.area.toLocaleString() 
+                               + ' square kilometers' : 'N/A'}</p>
+     <p>Languages: ${country.languages ? Object.
+                          values(country.languages).join(', ') : 'N/A'}</p>
+     <p>Capital: ${country.capital ? 
+            country.capital[0] : 'N/A'}</p>
+     <p>Timezones: ${country.timezones ? 
+                          country.timezones.join(', ') : 'N/A'}</p>
+     <p>Region: ${country.region ? 
+                            country.region : 'N/A'}</p>
+    `;
+    document.getElementById('country_details').innerHTML = details;
+    
+    return fetch('https://restcountries.com/v3.1/region/' + country.region);
```

Comment:
Do not create another fetch inside a then method.
https://docs.google.com/presentation/d/1iCoz17mBxZtIQ9PifAW0jjKsuv4xDTlx-TPQxdLySow/edit#slide=id.g2637aafa621_0_10
Click the link for reference